sonarqube/resource_sonarqube_user_token_test.go

```diff
@@ -48,3 +49,35 @@ func TestAccSonarqubeUserTokenBasic(t *testing.T) {
 		},
 	})
 }
+
+func testAccSonarqubeUserTokenExpirationDateConfig(rnd string, name string) string {
+	return fmt.Sprintf(`
+		resource "sonarqube_user" "%[1]s" {
+			login_name = "%[2]s"
+			name       = "%[2]s"
+			password   = "secret-sauce37!"
+		}
+		resource "sonarqube_user_token" "%[1]s" {
+			login_name      = sonarqube_user.%[1]s.login_name
+			name            = "%[2]s"
+			expiration_date = "%s"
+		}`, rnd, name, time.Now().AddDate(0, 0, 5).Format("2006-01-02"))
+}
+
+func TestAccSonarqubeUserTokenWithExpirationDate(t *testing.T) {
+	rnd := generateRandomResourceName()
+	name := "sonarqube_user_token." + rnd
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSonarqubeUserTokenExpirationDateConfig(rnd, "testAccSonarqubeUserTokenWithExpirationDate"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(name, "name", "testAccSonarqubeUserTokenWithExpirationDate"),
+				),
```

Comment:
Lets add another Check here for the presence of the token:

```go
resource.TestCheckResourceAttr(name, "expiration_date", "<VALUE>")
```

You'll have to pull the expiry time in testAccSonarqubeUserTokenExpirationDateConfig into a function parameter to reuse that value in  TestAccSonarqubeUserTokenWithExpirationDate.

Reply:
Hi it's done :)